jplag.frontend-utils/src/main/java/de/jplag/LanguageLoader.java

```diff
@@ -0,0 +1,79 @@
+package de.jplag;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.ServiceLoader;
+import java.util.TreeMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class contains methods to load {@link Language Languages}.
+ * @author Dominik Fuchss
+ */
+public final class LanguageLoader {
+    private static final Logger logger = LoggerFactory.getLogger(LanguageLoader.class);
+
+    private static Map<String, Language> loaded = null;
```

Comment:
Currently, the methods do not use the benefit of having the languages indexed by identifier.
I would propose to have a private method that loads all languages. This can either directly return the Map or guarantee that `loadedLanguages` (you did not rename it to my suggestion, any reason?) is populated after calling. Having the Map will simplify `getLanguage` (Map.get) and `getAllAvailableLanguageIdentifiers` (getKeySet). Furthermore, `getAllAvailableLanguages` does not have to duplicate the conversion from map to stream.

Reply:
Forgot to rename it :)